unity/ARUI/Assets/Scripts/ARUI/AngelARUI.cs

```diff
@@ -97,24 +99,24 @@ public void SetTasks(string[,] tasks)
     /// <summary>
     /// Set the callback function that is invoked if the user confirms the confirmation dialogue
     /// </summary>
-    public void SetUserIntentCallback(UnityAction userIntentCallBack) => onUserIntentConfirmedAction = userIntentCallBack;
+    public void SetUserIntentCallback(UnityAction<InterpretedAudioUserIntentMsg> userIntentCallBack) => onUserIntentConfirmedAction = userIntentCallBack;
 
     /// <summary>
-    /// If confirmation action is set - SetUserIntentCallback(...) - and no confirmation window is active at the moment, the user is shown a 
-    /// timed confirmation window. Recommended text: "Did you mean ...". If the user confirms the dialogue, the onUserIntentConfirmedAction action is invoked. 
+    /// If confirmation action is set - SetUserIntentCallback(...) - and no confirmation window is active at the moment, the user is shown a
+    /// timed confirmation window. Recommended text: "Did you mean ...". If the user confirms the dialogue, the onUserIntentConfirmedAction action is invoked.
     /// </summary>
     /// <param name="msg">message that is shown in the confirmation dialogue</param>
-    public void TryGetUserFeedbackOnUserIntent(string msg)
+    public void TryGetUserFeedbackOnUserIntent(InterpretedAudioUserIntentMsg intentMsg)
     {
         if (onUserIntentConfirmedAction == null || confirmationWindow != null) return;
 
         GameObject window = Instantiate(confirmationWindowPrefab, transform);
         confirmationWindow = window.AddComponent<ConfirmationDialogue>();
-        confirmationWindow.InitializeConfirmationNotification(msg, onUserIntentConfirmedAction);
+        confirmationWindow.InitializeConfirmationNotification(intentMsg, onUserIntentConfirmedAction);
```

Comment:
Maybe a @ovenmitt question: What happens if multiple user intents come in in quick succession such that this method (`TryGetUserFeedbackOnUserIntent`) is invoked again before the user has finished with the first confirmation? Are multiple confirmation windows created for the user to interact with?

Reply:
Right now, it is not designed for multiple user intents at the same time because, from the user's perspective, that is not something useful. Imagine you are talking to Siri, or any other person, really. There is usually one user intent at a time. 

Since it is _technically_ possible to get 'misfires' or 'multiple detected user intent within a short time', something in the system would have to make decisions about what user intent is the most important (or the correct one) and communicate that one to the user.

Naively, I could just store all incoming intents in a queue and show them the user in this order.

Reply:
OK, thanks for the input. Let's see how this implementation performs in combination with the audio intent components before we (I) get prematurely concerned about capability.